### PR TITLE
fix(argocd): remove dist-dynamic folder

### PR DESCRIPTION
**Pull request by karthikjeeyar**
## Description

This is a followup PR to  - https://github.com/janus-idp/backstage-showcase/pull/1697 and aims to load the argocd plugins correctly as a dynamic-plugin. This PR contains the following changes

1. removed the `dist-dynamic` folder as this is applicable only for the backend dynamic plugins.
2. Fixed the scalprum name and updated the correct argocd-plugin entry into the internal DynamicPlugin list.
3. updated the unit tests.

Remove already generated dist-dynamic folder

## Which issue(s) does this PR fix

BLOCKER ISSUE - https://issues.redhat.com/browse/RHIDP-4200 

**How to test:**

Add this in the dynamic-plugins.yaml configmap

```
  - package: ./dynamic-plugins/dist/backstage-community-plugin-redhat-argocd
    disabled: false


```

### CODE REVIEW

File: plugins/dynamic-plugins-info/src/components/InternalPluginsMap.tsx [hunk 1]
```diff
@@ -77,8 +77,8 @@ export const InternalPluginsMap: Record<string, string> = {
     './dynamic-plugins/dist/janus-idp-backstage-plugin-3scale-backend-dynamic',
   '@janus-idp/backstage-plugin-keycloak-backend-dynamic':
     './dynamic-plugins/dist/janus-idp-backstage-plugin-keycloak-backend-dynamic',
-  '@janus-idp/backstage-plugin-argocd':
-    './dynamic-plugins/dist/janus-idp-backstage-plugin-argocd',
+  'backstage-community-plugin-redhat-argocd':
+    './dynamic-plugins/dist/backstage-community-plugin-redhat-argocd',
```
Comment (karthikjeeyar) [inline plugins/dynamic-plugins-info/src/components/InternalPluginsMap.tsx:81]: I figured that, this entry has to be updated or else `PreInstalled` column will be set to `No` which is wrong.

Screenshot 2024-09-26 at 9 43 30 AM

After updating, it now shows the right value:

Screenshot 2024-09-26 at 10 24 13 AM

**Comment by karthikjeeyar** [comment]
I have verified this changes using the image `quay.io/janus-idp/backstage-showcase:pr-1699` in the cluster, Now it is dynamically loading the argocd plugin properly.

Overview page:

argocd_overview


CD tab:

cd_tab

**Comment by Fortune-Ndlovu** [comment]
lgtm

**Pull request merged**